server/py/framework/utils/clients/messaging.py

```diff
@@ -204,15 +345,32 @@ async def _on_request_failure(
         )
         if response.content:
             try:
-                data = await response.json()
-                error = data.get("error")
-                error_stack_trace = data.get("errorStackTrace")
+                response_body = await response.json()
+                error_details = response_body.get("detail", {})
             except Exception:
                 pass
             else:
-                log_kwargs.update(
-                    {"error": error, "error_stack_trace": error_stack_trace}
-                )
+                log_kwargs["error_details"] = error_details
+        logger.warning("Request to service failed", **log_kwargs)
+        if raise_on_failure:
+            mlrun.errors.raise_for_status(response)
+
+    @staticmethod
+    def _on_request_failure_sync(
+        method: str,
+        url: str,
+        response: requests.Response,
+        error_details: dict,
+        raise_on_failure: bool,
+        kwargs,
+    ):
+        log_kwargs = copy.deepcopy(kwargs)
+
+        # this can be big and spammy
+        log_kwargs.pop("json", None)
+        log_kwargs.update(
+            {"method": method, "url": url, "error_details": error_details}
+        )
```

Comment:
The difference here is only that async method requires `response_body = await response.json()` , 
suggestion - maybe most of the code can be moved to a common function?